### PR TITLE
Initial commit for logger app

### PR DESCRIPTION
**Pull request by soberpeach**
What does this PR do?
---------------------
Adds a "logger" datadog for use in e2e tests in the agent.

Which scenarios this will impact?
-------------------

Motivation
----------
Needed for e2e tests in the agent.

Additional Notes
----------------

### CODE REVIEW

File: components/datadog/apps/logger/k8s.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+					},
+				},
+				Spec: &corev1.PodSpecArgs{
+					Containers: corev1.ContainerArray{
+						corev1.ContainerArgs{
+							Name: pulumi.String("logger"),
+							// TODO Actual image name
+							Image: pulumi.String("PUT AN ACTUAL IMAGE HERE"),
+							Args: pulumi.StringArray{
+								pulumi.String("--loglevel"),
```
Comment (vboulineau) [inline components/datadog/apps/logger/k8s.go:67]: Not used

File: components/datadog/apps/logger/k8s.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+					Labels: pulumi.StringMap{
+						"app": pulumi.String("logger"),
+					},
+				},
+				Spec: &corev1.PodSpecArgs{
+					Containers: corev1.ContainerArray{
+						corev1.ContainerArgs{
+							Name: pulumi.String("logger"),
+							// TODO Actual image name
+							Image: pulumi.String("PUT AN ACTUAL IMAGE HERE"),
```
Comment (vboulineau) [inline components/datadog/apps/logger/k8s.go:65]: You need to put the GHCR image here.

File: components/datadog/apps/logger/images/logger/main.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	"io"
+	"net/http"
+	"os"
+)
+
+func main() {
+	handler := http.HandlerFunc(handleRequest)
+	http.Handle("/", handler)
+
+	err := http.ListenAndServe(":3333", nil)
```
Comment (vboulineau) [inline components/datadog/apps/logger/images/logger/main.go:14]: How do you plan to reach `3333`?

File: components/datadog/apps/logger/k8s.go (hunk 1 above)
Comment (soberpeach) [reply components/datadog/apps/logger/k8s.go]: Removed

File: components/datadog/apps/logger/k8s.go (hunk 2 above)
Comment (soberpeach) [reply components/datadog/apps/logger/k8s.go]: Added!

File: components/datadog/apps/logger/images/logger/main.go (hunk 3 above)
Comment (soberpeach) [reply components/datadog/apps/logger/images/logger/main.go:14]: I plan to query it from an e2e test in the datadog agent! I've extracted it to a variable

File: components/datadog/apps/logger/k8s.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+				Metadata: &metav1.ObjectMetaArgs{
+					Labels: pulumi.StringMap{
+						"app": pulumi.String("logger"),
+					},
+				},
+				Spec: &corev1.PodSpecArgs{
+					Containers: corev1.ContainerArray{
+						corev1.ContainerArgs{
+							Name:  pulumi.String("logger"),
+							Image: pulumi.String("http://ghcr.io/datadog/apps-logger:main"),
```
Comment (KevinFairise2) [inline components/datadog/apps/logger/k8s.go:66]: You must not include the `http://` in the name of the image
```suggestion
							Image: pulumi.String("ghcr.io/datadog/apps-logger:main"),
```

File: components/datadog/apps/logger/k8s.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+							},
+						},
+					},
+				},
+			},
+		},
+	}, opts...); err != nil {
+		return nil, err
+	}
+
```
Comment (KevinFairise2) [inline components/datadog/apps/logger/k8s.go:92]: To be able to easily send an http request from another pod within the cluster I'd suggest adding a service attached to your deployment

```
	if _, err := corev1.NewService(e.Ctx, "logger", &corev1.ServiceArgs{
		Metadata: &metav1.ObjectMetaArgs{
			Name:      pulumi.String("logger"),
			Namespace: pulumi.String(namespace),
		},
		Spec: &corev1.ServiceSpecArgs{
			Selector: pulumi.StringMap{
				"app": pulumi.String("logger"),
			},
			Ports: &corev1.ServicePortArray{
				&corev1.ServicePortArgs{
					Port:       pulumi.Int(LoggerPort),
					TargetPort: pulumi.Int(LoggerPort),
				},
			},
		},
	}, opts...); err != nil {
		return nil, err
	}
```

Comment (soberpeach) [reply components/datadog/apps/logger/k8s.go:92]: Added! Thanks for this.

**Review by KevinFairise2 — APPROVED** [review]

**Pull request merged**